packages/core/react/src/WalletProvider.tsx

```diff
@@ -23,15 +23,13 @@ export interface WalletProviderProps {
 }
 
 const initialState: {
-    wallet: EnhancedWallet | null;
+    wallet: Wallet | null;
```

Comment:
wdyt, can we roll the base Wallet change into this? I figure if we're going to make that change we might as well do it now.

Reply:
This PR's getting a bit out of control already. How about we bring back `EnhancedWallet`/`WalletWithReadyState` now, and then dissolve the whole thing back into `WalletAdapter` in a future one?

Reply:
Works for me!